pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -907,8 +907,9 @@ func autoAllocateIPs(services []*model.Service) []*model.Service {
 		//   for NONE because we will not know the original DST IP that the application requested.
 		// 2. the address is not set (0.0.0.0)
 		// 3. the hostname is not a wildcard
-		if svc.DefaultAddress == constants.UnspecifiedIP && !svc.Hostname.IsWildCarded() &&
-			svc.Resolution != model.Passthrough {
+		// TODO(jaellio): Add a feature flag to conditionally wildcard hostnames if the service entry
```

Comment:
We should also add this conditional before merging this

Reply:
Not adding support here since I believe this is deprecated.